src/main/java/wolox/training/controllers/BookController.java

```diff
@@ -70,10 +75,55 @@ public String greeting(@RequestParam(name = "name", required = false, defaultVal
         return bookRepository.findByTitle(bookTitle);
     }
 
-    @GetMapping("/publisherGenreYear/{publisher}/{genre}/{year}")
-    public List<Book> findByGenrePublisherYear(@PathVariable String publisher, @PathVariable String genre,
-            @PathVariable String year) {
-        return bookRepository.findByPublisherAndGenreAndYear(publisher, genre, year);
+    @GetMapping("/publisherGenreYear")
```

Comment:
you can delete this endpoint and use the @GetMapping("/allBooks") that endpoint uses the same params of this

Reply:
Fixed at last commit!